src/DistributedLock.Tests/Tests/Etcd/EtcdSynchronizationProviderTest.cs

```diff
@@ -0,0 +1,35 @@
+﻿using Medallion.Threading.Etcd;
+using Medallion.Threading.Tests.Etcd;
+using NUnit.Framework;
+
+namespace Medallion.Threading.Tests.Etcd;
+
+
+[Category("CI")]
+public class EtcdSynchronizationProviderTest
+{
+
+    private readonly EtcdClusterSetup _etcdClusterSetup = new();
+
+    [OneTimeSetUp]
+    public async Task OneTimeSetUp()
+    {
+        await this._etcdClusterSetup.ClusterSetup();
+    } 
```

Comment:
Look to how other providers handle one-time setup and tear down without requiring every test to do it.